### DISCUSSION TITLE
es/guides/triton-inference-server/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#14641] [Docs]
# es/guides/triton-inference-server/

Aprende a integrar Ultralytics YOLOv8  con NVIDIA Triton  Inference Server para un despliegue de modelos de IA escalable y de alto rendimiento.

https://docs.ultralytics.com/es/guides/triton-inference-server/

**Comment by wisrovi** [comment]
Cordial saludo

tengo un modelo rtdetr-x.pt de https://docs.ultralytics.com/es/models/rtdetr/#usage-examples

lo he convertido a onnx con https://docs.ultralytics.com/es/integrations/onnx/#installation y luego intente montarlo con triton en un server, al parecer dice estar funcionando el servicio, pero no logro realizar ninguna inferencia, 

estoy en ubuntu 22.04 con Intel I5 y NVIDIA 1060Ti

he instalado triton con:
pip install tritonclient[all]                                                                             

pero me ha dado el error:
zsh: no matches found: tritonclient[all]

asi que intente con: pip install tritonclient

pero creo que alli estuvo el problema, pero no se como resolverlo, me ayudarian por favor?

**Reply by glenn-jocher** [reply]
@wisrovi hola,

Parece que el problema radica en la instalación de `tritonclient`. El error "zsh: no matches found: tritonclient[all]" sugiere que el shell zsh está interpretando incorrectamente los corchetes. Puedes intentar instalar `tritonclient` usando comillas para evitar este problema:

```sh
pip install 'tritonclient[all]'
```

Si esto no resuelve el problema, asegúrate de que tu entorno de Python esté correctamente configurado y que Docker esté instalado y funcionando. Una vez que `tritonclient` esté instalado correctamente, sigue los pasos de la documentación para configurar y ejecutar Triton Inference Server.

Para más detalles, puedes consultar la guía completa en [Triton Inference Server con Ultralytics YOLOv8](https://docs.ultralytics.com/guides/triton-inference-server/).

Saludos.

**Reply by wisrovi** [reply]
gracias, con eso lo pude instalar, ahora me estoy peleando con la conversion a ONNX porque el archivo se convierte pero no me da rsultados de inferencia cercanos a los resultados antes de la conversion, pero postee este problema en https://docs.ultralytics.com/es/integrations/onnx/#how-can-i-troubleshoot-issues-when-exporting-yolov8-models-to-onnx

**Reply by glenn-jocher** [reply]
@wisrovi me alegra saber que pudiste instalar `tritonclient`. Sobre la conversión a ONNX y los resultados de inferencia, es importante asegurarse de que todos los parámetros y configuraciones del modelo se mantengan consistentes durante la exportación. A veces, diferencias en la precisión o en la configuración del modelo pueden afectar los resultados. Te recomiendo revisar la documentación de ONNX y Triton para asegurarte de que todos los pasos se han seguido correctamente. Además, verifica si hay actualizaciones disponibles para las herramientas que estás utilizando. Si el problema persiste, podrías intentar ajustar los parámetros de exportación o consultar con la comunidad para obtener más sugerencias.
